### PR TITLE
clisp: 2.49 -> 2.50pre2023-01-12; 2.50pre20171114 -> remove

### PR DESCRIPTION
**Pull request by tobim**
###### Description of changes

Update clisp to new active repo at https://gitlab.com/gnu-clisp/clisp. 
Using the latest commit on the 2.50 branch as version.
This goes towards #87909.

I also added darwin to the platforms again, but can't test it. Help would be greatly appreciated!

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by tobim** [comment]
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>25 packages built:</summary>
  <ul>
    <li>advi</li>
    <li>apostrophe</li>
    <li>avrdude</li>
    <li>avrdudess</li>
    <li>blahtexml</li>
    <li>blahtexml.doc</li>
    <li>bluespec</li>
    <li>bluespec.doc</li>
    <li>clisp</li>
    <li>klipper-firmware</li>
    <li>klipper-flash</li>
    <li>ne</li>
    <li>nuweb</li>
    <li>opentx</li>
    <li>qmk</li>
    <li>qmk.dist</li>
    <li>texlive.combined.scheme-full</li>
    <li>texlive.combined.scheme-medium</li>
    <li>tinygo</li>
    <li>xyce</li>
    <li>xyce-parallel</li>
    <li>xyce-parallel.doc</li>
    <li>xyce.doc</li>
    <li>yosys-bluespec</li>
    <li>zettlr</li>
  </ul>
</details>

File: pkgs/development/interpreters/clisp/default.nix [hunk 1]
```diff
@@ -74,35 +72,26 @@ stdenv.mkDerivation rec {
...
-  doCheck = false;
-
   meta = {
     description = "ANSI Common Lisp Implementation";
     homepage = "http://clisp.cons.org";
     maintainers = lib.teams.lisp.members;
-    platforms = lib.platforms.unix;
-    # problems on Darwin: https://github.com/NixOS/nixpkgs/issues/20062
-    broken = stdenv.hostPlatform.isDarwin || stdenv.hostPlatform.isAarch64;
-    license = lib.licenses.gpl2;
```
Comment (nagy) [inline pkgs/development/interpreters/clisp/default.nix:106]: why remove the license ?

Comment (tobim) [reply pkgs/development/interpreters/clisp/default.nix:106]: That was not intentional. Thanks for spotting!
I changed it to `gpl2Plus` according to https://gitlab.com/gnu-clisp/clisp/-/blob/master/COPYRIGHT.

**Comment by 7c6f434c** [comment]
Looks like (at least) OpenGL part of CLX does not build on macOS? Although this probably means that the main console part does build.

Separately: if you consider this a replacement for the main line, maybe fix the error of my ways six years ago and use `2.49-unstable-2023-01-09` instead of a `2.50pre`-based version? (Using past versions instead of prediceted future ones for snapshots without upstream versions, as has been argued in the last couple of years)

**Comment by tobim** [comment]
> Looks like (at least) OpenGL part of CLX does not build on macOS? Although this probably means that the main console part does build.

I finally got temporary access to a darwin machine and was able to fix the build. Please check the latest commit.

> Separately: if you consider this a replacement for the main line, maybe fix the error of my ways six years ago and use `2.49-unstable-2023-01-09` instead of a `2.50pre`-based version? (Using past versions instead of prediceted future ones for snapshots without upstream versions, as has been argued in the last couple of years)

In this case we're building of the `2.50` branch though, so I'd argue that naming is more correct than `2.49-unstable`.

**Comment by tobim** [comment]
Result of `nixpkgs-review pr 236927` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>xyce</li>
    <li>xyce-parallel</li>
    <li>xyce-parallel.doc</li>
    <li>xyce.doc</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>advi</li>
    <li>blahtexml</li>
    <li>blahtexml.doc</li>
    <li>clisp</li>
    <li>ne</li>
    <li>nuweb</li>
    <li>texlive.combined.scheme-full</li>
    <li>texlive.combined.scheme-medium</li>
  </ul>
</details>

**Comment by tobim** [comment]
Result of `nixpkgs-review pr 236927` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>klipper-firmware</li>
    <li>klipper-flash</li>
  </ul>
</details>
<details>
  <summary>24 packages built:</summary>
  <ul>
    <li>advi</li>
    <li>apostrophe</li>
    <li>avrdude</li>
    <li>avrdudess</li>
    <li>blahtexml</li>
    <li>blahtexml.doc</li>
    <li>bluespec</li>
    <li>bluespec.doc</li>
    <li>clisp</li>
    <li>ne</li>
    <li>nuweb</li>
    <li>opentx</li>
    <li>qmk</li>
    <li>qmk.dist</li>
    <li>texlive.combined.scheme-full</li>
    <li>texlive.combined.scheme-medium</li>
    <li>tinygo</li>
    <li>xyce</li>
    <li>xyce-parallel</li>
    <li>xyce-parallel.doc</li>
    <li>xyce.doc</li>
    <li>yosys-bluespec</li>
    <li>zettlr</li>
    <li>zettlr-beta</li>
  </ul>
</details>

klipper-firmware is failing on hydra with the same error: https://hydra.nixos.org/eval/1799650?filter=klipper-firmware&compare=1799628&full=

**Comment by 7c6f434c** [comment]
For upstreams who do more releases, I would actually expect the branch to be renamed to 3.0 or 2.49.1 like nothing has happened, but yeah probably `clisp` will do the simplest thing available with releases (if at all)

**Pull request merged**